base/doc/ltnews38.tex

```diff
@@ -278,67 +278,149 @@ \section{Hooks, sockets and plugs}
+%to record, in addition, both a title (such as the text used in a
+%section head) and the \emph{logical name} used
+%for an associated link target
+%%FMi new
+%if these have been set by packages such as  \pkg{nameref} or \pkg{hyperref}.
+
+%%FMi: meanwhile sounds as if it just happened after 2023 Spring ...
+%%     So moved the above text into a footnote (is anyway old news) and
+%%    did some rewrites below.
+%%
+%Meanwhile, other packages had been extending
+%this \enquote{label-ref system} in various ways.
+
+Over the years \LaTeX{} packages appeared that extended
+this basic \enquote{label-ref system} in various ways.
+%
+For example, the \pkg{refcount} package made a small but significant change
+to the functions used to access recorded values, by making them expandable.
+And the \pkg{smart-ref} package supports the storage of a larger
+collection of counter values so that, for example, a cross-reference
+can refer to the relevant chapter together with an equation tag.
+The \pkg{cleveref} package stores (by means of a second, internal
+\enquote{logical label}) extra information such as the name of the counter.
+The \pkg{hyperref} package adds the \cs{autoref} command, which
+tries to retrieve the name of a counter from the  \emph{logical name}
+used for a link target.
+The \pkg{tikzmarks} package records information about
+ \emph{labelled positions} in pictures when using \pkg{tikz}.
+Finally, the \pkg{zref} package implements many related ideas, including a general
+idea of properties and lists of properties, with methods to record, and
+subsequently access, the value of any declared property.
+
+
+%%FMi some rewrite:
+Starting with this release, the \LaTeX{} kernel now offers the handling general document
+properties as a core functionality with standard interfaces.
+%
+This is based on concepts introduced by the \pkg{zref}
+package but with some differences in detail, particularly
+in the implementation.
+It supports the declaration of new properties, and the recording of
+the values of any list of properties. These values are retrieved expandably.
+
+To set up a new property that is the current chapter
+number, for example, here is the declaration to use.
 \begin{verbatim}
-\NewProperty{chapter}{now}{?}{\thechapter}
+  \NewProperty{chapter}{now}{?}{\thechapter}
 \end{verbatim}
-The second argument means that the property is recorded immediately 
-(\enquote{now}) and not at \cs{shipout}. The third argument sets a default if a 
-label is unknown. It is then possible to record a list of properties with 
-\cs{RecordProperties}: 
+The second argument means that the property value will be recorded
+immediately (\enquote{now}), and not \enquote{during the next \cs{shipout}}.
+The third argument sets a default
+to be used when, for example, an unknown label is supplied.
+%%FMi added
+The final argument contains the code that produces the current
+property value when when recording the value is requested.
```

Comment:
when when -> when

Reply:
Suggestions:

. . . the code that will, as part of the recording process, be expanded to obtain the value to record for this property.